.misc/check_setuptools.py

```diff
@@ -2,14 +2,16 @@
 
 import subprocess
 import sys
+import re
 
 import setuptools
 
 
 def get_setuptools_version():
     with open('requirements.txt') as f:
         for line in f:
-            if line.startswith('setuptools'):
+            # Ensure that line matched is setuptools
+            if re.match(r'^setuptools(>|<|~|=|$)', line):
```

Comment:
`re.compile` is better option since we need to match it multiple times and regex remains same ever time.